src/item.cpp

```diff
@@ -413,6 +413,26 @@ static const item *get_most_rotten_component( const item &craft )
     return most_rotten;
 }
 
+// sorts with localized_compare, and enumerates entries, if more than \p max entries
+// the rest are abbreviated into " and %d more"
+static std::string enumerate_lcsorted_with_limit( const std::vector<std::string> &v, int max = 5 )
+{
+    std::vector<std::string> c = v; // copy so we can non destructively sort
+    std::sort( c.begin(), c.end(), localized_compare );
+    if( max < 1 ) {
+        debugmsg( "max expected to be 1 or more" );
+        return "";
+    }
+    if( max >= static_cast<int>( c.size() ) ) {
+        return enumerate_as_string( c );
+    }
+    const std::string res =  enumerate_as_string( c,
+    [&max]( const std::string & r ) {
+        return ( --max >= 0 ) ? r : "";
+    }, enumeration_conjunction::none );
+    return res + string_format( _( " and %d more" ), -max );
+}
```

Comment:
```suggestion
static std::string enumerate_lcsorted_with_limit( const std::vector<std::string> &v, const int max = 5 )
{
    if( max < 1 ) {
        debugmsg( "Expected max to be 1 or more" );
        return std::string();
    }
    std::vector<std::string> c( std::min<size_t>( max, v.size() ) );
    std::partial_sort_copy( v.begin(), v.end(), c.begin(), c.end(), localized_compare );
    const int more = static_cast<int>( v.size() - c.size() );
    if( more == 0 ) {
        return enumerate_as_string( c );
    }
    const std::string res = enumerate_as_string( c, enumeration_conjunction::none );
    return string_format( max == 1 ? _( "%s and %d more" ) : _( "%s, and %d more" ), res, more );
}
```
Should be more memory and time efficient, please test if it works.